src/screen/dashboard.rs

```diff
@@ -887,18 +902,39 @@ impl Dashboard {
                         }
                     }
                     ToggleNicklist => {
-                        if let Some((_, _, pane)) = self.get_focused_mut(main_window) {
-                            pane.update_settings(|settings| {
-                                settings.channel.nicklist.enabled =
-                                    !settings.channel.nicklist.enabled
+                        if let Some((_, pane, buffer_settings)) =
+                            self.get_focused_with_buffer_settings_mut(main_window)
+                        {
+                            let buffer = pane.buffer.data();
+
+                            let settings = pane.update_settings(|settings| {
+                                settings.channel.nicklist.toggle_visibility()
                             });
+
+                            if let Some(buffer) = buffer {
+                                buffer_settings.insert(buffer, settings);
+                            }
+
+                            self.last_changed = Some(Instant::now());
+                            return (self.save_dashboard(), None);
                         }
                     }
                     ToggleTopic => {
-                        if let Some((_, _, pane)) = self.get_focused_mut(main_window) {
-                            pane.update_settings(|settings| {
-                                settings.channel.topic.enabled = !settings.channel.topic.enabled
+                        if let Some((_, pane, buffer_settings)) =
+                            self.get_focused_with_buffer_settings_mut(main_window)
+                        {
+                            let buffer = pane.buffer.data();
+
+                            let settings = pane.update_settings(|settings| {
+                                settings.channel.topic.toggle_visibility()
                             });
+
+                            if let Some(buffer) = buffer {
+                                buffer_settings.insert(buffer, settings);
+                            }
+
+                            self.last_changed = Some(Instant::now());
+                            return (self.save_dashboard(), None);
```

Comment:
We shouldn't need to manually save the dashboard, this is what `last_changed` is responsible over. We save dashboard during tick if that's set. So we will be saving it twice now.